src/lib.rs

```diff
@@ -0,0 +1,371 @@
+//! Entry point into the task generation logic.
+//!
+//! This code will go through the entire evergreen configuration and create task definitions
+//! for any tasks that need to be generated. It will then add references to those generated
+//! tasks to any build variants to expect to run them.
+#![cfg_attr(feature = "strict", deny(missing_docs))]
+
+use std::{
+    collections::{HashMap, HashSet},
+    path::Path,
+    sync::{Arc, Mutex},
+};
+
+use anyhow::Result;
+use evg_config::{EvgConfigService, EvgConfigUtils, EvgConfigUtilsImpl, EvgProjectConfig};
+use lazy_static::lazy_static;
+use regex::Regex;
+use resmoke_proxy::ResmokeProxy;
+use shrub_rs::models::{
+    project::EvgProject,
+    task::{EvgTask, TaskRef},
+    variant::{BuildVariant, DisplayTask},
+};
+use task_name::remove_gen_suffix;
+use task_types::{
+    fuzzer_tasks::{FuzzerGenTaskParams, GenFuzzerService, GenFuzzerServiceImpl},
+    generated_suite::GeneratedSuite,
+    multiversion::MultiversionServiceImpl,
+};
+use tracing::{event, Level};
+
+mod evergreen_names;
+mod evg_config;
+mod resmoke_proxy;
+mod task_name;
+mod task_types;
+
+/// Directory to store the generated configuration in.
+const CONFIG_DIR: &str = "generated_resmoke_config";
+
+lazy_static! {
```

Comment:
`lazy_static` is a macro that will let you define static values, but they won't actually be created until they first get used.